Source/WebKit/GPUProcess/RemoteSharedResourceCache.cpp

```diff
@@ -57,6 +71,29 @@ void RemoteSharedResourceCache::releaseSerializedImageBuffer(WebCore::RenderingR
     m_serializedImageBuffers.remove({ { identifier, 0 }, 0 });
 }
 
+void RemoteSharedResourceCache::didAddAcceleratedImageBuffer()
+{
+    ++acceleratedImageBufferGlobalCount;
+    ++m_acceleratedImageBufferCount;
+}
+
+void RemoteSharedResourceCache::didTakeAcceleratedImageBuffer()
+{
+    --acceleratedImageBufferGlobalCount;
+    --m_acceleratedImageBufferCount;
+}
+
+WebCore::RenderingMode RemoteSharedResourceCache::adjustAcceleratedImageBufferRenderingMode(RenderingPurpose renderingPurpose) const
+{
+    if (acceleratedImageBufferGlobalCount >= acceleratedImageBufferGlobalLimit)
```

Comment:
Could be worth a comment that this is known to be racy, but that it shouldn't matter in practice.